src/utils.zig

```diff
@@ -1907,6 +1907,155 @@ test "read/write mixed" {
 	try std.testing.expect(reader.remaining.len == 0);
 }
 
+pub fn SparseSet(comptime T: type, comptime IdType: type) type { // MARK: SparseSet
+	std.debug.assert(@typeInfo(IdType) == .int);
+	std.debug.assert(@typeInfo(IdType).int.signedness == .unsigned);
+
+	return struct {
+		const Index = enum(IdType) {
+			noValue = std.math.maxInt(IdType),
+			_,
+		};
+
+		const Self = @This();
+
+		dense: main.ListUnmanaged(T) = .{},
+		reverseIds: main.ListUnmanaged(IdType) = .{},
+		sparse: main.ListUnmanaged(Index) = .{},
```

Comment:
Maybe we could call them denseToSparseIndex and sparseToDenseIndex?

Reply:
There is also no tests covering that while there should be one since this is a very important edge case.